utils/global-state-update-gen/src/generic/testing.rs

```diff
@@ -849,6 +947,103 @@ fn should_replace_a_delegator() {
     assert_eq!(update.len(), 6);
 }
 
+#[test]
+fn should_replace_a_delegator_with_unbonding() {
+    let mut rng = TestRng::new();
+
+    let validator1 = PublicKey::random(&mut rng);
+    let delegator1 = PublicKey::random(&mut rng);
+    let delegator2 = PublicKey::random(&mut rng);
+
+    let mut reader = MockStateReader::new().with_validators(
+        vec![(
+            validator1.clone(),
+            U512::from(101),
+            ValidatorConfig {
+                bonded_amount: U512::from(101),
+                delegation_rate: Some(5),
+                delegators: Some(vec![DelegatorConfig {
+                    public_key: delegator1.clone(),
+                    delegated_amount: U512::from(13),
+                }]),
+            },
+        )],
+        &mut rng,
+    );
+
+    // we'll be replacing the delegator
+    let config = Config {
+        accounts: vec![AccountConfig {
+            public_key: validator1.clone(),
+            balance: Some(U512::from(101)),
+            validator: Some(ValidatorConfig {
+                bonded_amount: U512::from(101),
+                delegation_rate: None,
+                delegators: Some(vec![DelegatorConfig {
+                    public_key: delegator2.clone(),
+                    delegated_amount: U512::from(14),
+                }]),
+            }),
+        }],
+        only_listed_validators: false,
+        slash_instead_of_unbonding: false,
+        ..Default::default()
+    };
+
+    let update = get_update(&mut reader, config);
+
+    update.assert_seigniorage_recipients_written(&mut reader);
+    update.assert_total_supply(&mut reader, 229);
```

Comment:
Just to make sure I understand correctly: The total supply before the update was 215, right?

Reply:
Right :+1: